gnomad_qc/v4/create_release/create_combined_faf_release_ht.py

```diff
@@ -0,0 +1,455 @@
+        - freq_meta: {prefix}_freq_meta
+        - faf_meta: {prefix}_faf_meta
+
+    This only keeps variants that PASS filtering.
+
+    :param ht: Table with frequency and FAF information.
+    :param pops: List of populations to include in the reduced frequency arrays.
+    :param faf_pops: List of populations to include in the reduced FAF arrays.
+    :param prefix: Prefix to add to each of the filtered annotations.
+    :return: Table with filtered frequency and FAF information.
+    """
+    logger.info("Filtering Table to only variants that are PASS...")
+    ht = ht.filter(hl.len(ht.filters) == 0)
+
+    def _get_pop_meta_indices(
+        meta: hl.ArrayExpression, pop_list: List[str]
+    ) -> Tuple[List[int], List[dict]]:
+        """
+        Keep full gnomAD callset adj [0], raw [1], and ancestry-specific adj frequencies.
+
+        :param meta: Array of frequency metadata.
+        :param pop_list: List of populations to keep.
+        :return: Indices of populations to keep and their metadata.
+        """
+        meta = hl.eval(meta)
+        idx = [0, 1]
+        idx.extend(
+            [i for i, m in enumerate(meta) if m.get("pop") in pop_list and len(m) == 2]
+        )
+        meta = [meta[i] for i in idx]
+
+        return idx, meta
+
+    logger.info("Keeping only frequencies that are needed (adj, raw, adj by pop)...")
+    freq_idx, freq_meta = _get_pop_meta_indices(ht.freq_meta, pops)
+    faf_idx, faf_meta = _get_pop_meta_indices(ht.faf_meta, faf_pops)
+
+    # Rename filtered annotations with supplied prefix.
+    ht = ht.select(
+        **{
+            f"{prefix}_freq": [ht.freq[i] for i in freq_idx],
+            f"{prefix}_faf": [ht.faf[i] for i in faf_idx],
+            f"{prefix}_grpmax": ht.grpmax,
+        }
+    )
+    ht = ht.select_globals(
+        **{
+            f"{prefix}_freq_meta": freq_meta,
+            f"{prefix}_faf_meta": faf_meta,
+        }
+    )
+
+    return ht
+
+
+def get_joint_freq_and_faf(
+    genomes_ht: hl.Table,
+    exomes_ht: hl.Table,
+    faf_pops_to_exclude: Set[str] = POPS_TO_REMOVE_FOR_POPMAX,
+) -> hl.Table:
+    """
+    Get joint genomes and exomes frequency and FAF information.
+
+    :param genomes_ht: Table with genomes frequency and FAF information.
+    :param exomes_ht: Table with exomes frequency and FAF information.
+    :param faf_pops_to_exclude: Set of populations to exclude from the FAF calculation.
+    :return: Table with joint genomes and exomes frequency and FAF information.
+    """
+    logger.info("Performing an inner join on frequency HTs...")
+    ht = genomes_ht.join(exomes_ht, how="inner")
+
+    # Merge exomes and genomes frequencies.
+    freq, freq_meta = merge_freq_arrays(
+        farrays=[ht.genomes_freq, ht.exomes_freq],
+        fmeta=[ht.genomes_freq_meta, ht.exomes_freq_meta],
+    )
+    freq_meta = hl.literal(freq_meta)
+
+    # Compute FAF on the merged exomes + genomes frequencies.
```

Comment:
can we also add the "popmax/grpmax faf" annotation to the Table rather than having to recalculate it if we ever want to pull out what's on the browser, such as for
 'genomes_faf': array<struct {
        faf95: float64, 
        faf99: float64
    }> 
    
    add 'grpmax' under genomes_faf.faf95 and genomes_faf.faf99
    
for the vrs functions we're adding something similar to what was done for the browser:
    https://github.com/broadinstitute/gnomad-browser/blob/7924d1fdbaac860fa490920575d1013559f0b192/data-pipeline/src/data_pipeline/datasets/gnomad_v3/gnomad_v3_variants.py#L165

Reply:
I addressed both your comments in PR #446.